### PR TITLE
Implement azure redis cluster auto config

### PR DESCRIPTION
**Pull request by warrenzhu25**

### CODE REVIEW

File: spring-cloud-azure-autoconfigure/src/main/java/com/microsoft/azure/spring/cloud/autoconfigure/storage/AzureStorageAutoConfiguration.java [hunk 1]
```diff
@@ -15,8 +15,8 @@
 import com.microsoft.azure.spring.cloud.context.core.AzureUtil;
 import com.microsoft.azure.spring.cloud.storage.AzureStorageProtocolResolver;
 import com.microsoft.azure.storage.CloudStorageAccount;
-import org.apache.commons.logging.Log;
-import org.apache.commons.logging.LogFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
```
Comment (sophiaso) [inline spring-cloud-azure-autoconfigure/src/main/java/com/microsoft/azure/spring/cloud/autoconfigure/storage/AzureStorageAutoConfiguration.java]: This change is already in master branch, but the diff is still here.

File: spring-cloud-azure-context/src/main/java/com/microsoft/azure/spring/cloud/context/core/AzureAdmin.java [hunk 2]
```diff
@@ -144,7 +147,13 @@ public ServiceBusNamespace getOrCreateServiceBusNamespace(String namespace) {
     }
 
     public ServiceBusNamespace getServiceBusNamespace(String namespace) {
-        return azure.serviceBusNamespaces().getByResourceGroup(resourceGroup, namespace);
+        try {
+            return azure.serviceBusNamespaces().getByResourceGroup(resourceGroup, namespace);
+        } catch (NullPointerException ignore) {
+            // azure management api has no way to determine whether an service bus namespace exists
+            // Workaround for this is by catching NPE
+            return null;
```
Comment (sophiaso) [inline spring-cloud-azure-context/src/main/java/com/microsoft/azure/spring/cloud/context/core/AzureAdmin.java]: Do we need add any log here?

File: spring-cloud-azure-autoconfigure/src/main/java/com/microsoft/azure/spring/cloud/autoconfigure/storage/AzureStorageAutoConfiguration.java [hunk 3]
```diff
@@ -43,7 +43,7 @@
 @EnableConfigurationProperties(AzureStorageProperties.class)
 @Import(AzureStorageProtocolResolver.class)
 public class AzureStorageAutoConfiguration {
-    private static final Log LOGGER = LogFactory.getLog(AzureStorageAutoConfiguration.class);
+    private static final Logger LOGGER = LoggerFactory.getLogger(AzureStorageAutoConfiguration.class);
```
Comment (Incarnation-p-lee) [inline spring-cloud-azure-autoconfigure/src/main/java/com/microsoft/azure/spring/cloud/autoconfigure/storage/AzureStorageAutoConfiguration.java]: try `@Slf4j` from lombok

File: spring-cloud-azure-autoconfigure/src/main/java/com/microsoft/azure/spring/cloud/autoconfigure/storage/AzureStorageAutoConfiguration.java (hunk 1 above)
Comment (warrenzhu25) [reply spring-cloud-azure-autoconfigure/src/main/java/com/microsoft/azure/spring/cloud/autoconfigure/storage/AzureStorageAutoConfiguration.java]: Yes. It seems kind of github bug.

File: spring-cloud-azure-context/src/main/java/com/microsoft/azure/spring/cloud/context/core/AzureAdmin.java (hunk 2 above)
Comment (warrenzhu25) [reply spring-cloud-azure-context/src/main/java/com/microsoft/azure/spring/cloud/context/core/AzureAdmin.java]: Might not. NPE really means namespace not exist. So just returning null.

File: spring-cloud-azure-autoconfigure/src/main/java/com/microsoft/azure/spring/cloud/autoconfigure/storage/AzureStorageAutoConfiguration.java (hunk 3 above)
Comment (warrenzhu25) [reply spring-cloud-azure-autoconfigure/src/main/java/com/microsoft/azure/spring/cloud/autoconfigure/storage/AzureStorageAutoConfiguration.java]: Will change it in another PR.

**Pull request merged**